qt/scientific_interfaces/Indirect/IndirectSymmetriseView.cpp

```diff
@@ -0,0 +1,434 @@
+  // opposite X value
+  // i.e. RS min is X max
+
+  auto positiveERaw = m_uiForm.ppRawPlot->addRangeSelector("PositiveE");
+  positiveERaw->setColour(Qt::darkGreen);
+
+  // SIGNAL/SLOT CONNECTIONS
+  // Validate the E range when it is changed
+
+  connect(m_dblManager, SIGNAL(valueChanged(QtProperty *, double)), this, SIGNAL(valueChanged(QtProperty *, double)));
+  // Plot miniplot when file has finished loading
+  connect(m_uiForm.dsInput, SIGNAL(dataReady(QString const &)), this, SIGNAL(dataReady(QString const &)));
+  // Preview symmetrise
+  connect(m_uiForm.pbPreview, SIGNAL(clicked()), this, SIGNAL(previewClicked()));
+  // X range selectors
+  connect(positiveERaw, SIGNAL(minValueChanged(double)), this, SLOT(xRangeMinChanged(double)));
+  connect(positiveERaw, SIGNAL(maxValueChanged(double)), this, SLOT(xRangeMaxChanged(double)));
+  // Handle running, plotting and saving
+  connect(m_uiForm.pbRun, SIGNAL(clicked()), this, SIGNAL(runClicked()));
+  connect(m_uiForm.pbSave, SIGNAL(clicked()), this, SIGNAL(saveClicked()));
+
+  connect(this, SIGNAL(updateRunButton(bool, std::string const &, QString const &, QString const &)), this,
+          SLOT(updateRunButton(bool, std::string const &, QString const &, QString const &)));
+}
+
+//----------------------------------------------------------------------------------------------
+/** Destructor
+ */
+IndirectSymmetriseView::~IndirectSymmetriseView() {}
+
+void IndirectSymmetriseView::setDefaults() {
+  // Set default X range values
+  m_dblManager->setValue(m_properties["EMax"], 0.5);
+  m_dblManager->setValue(m_properties["EMin"], 0.1);
+
+  // Set default x axis range
+  QPair<double, double> defaultRange(-1.0, 1.0);
+  m_uiForm.ppRawPlot->setAxisRange(defaultRange, AxisID::XBottom);
+  m_uiForm.ppPreviewPlot->setAxisRange(defaultRange, AxisID::XBottom);
+
+  // Disable run until preview is clicked
+  m_uiForm.pbRun->setEnabled(false);
+  m_uiForm.pbPreview->setEnabled(false);
+
+  // Allows empty workspace selector when initially selected
+  m_uiForm.dsInput->isOptional(true);
+
+  // Disables searching for run files in the data archive
+  m_uiForm.dsInput->isForRunFiles(false);
+}
+
+IndirectPlotOptionsView *IndirectSymmetriseView::getPlotOptions() { return m_uiForm.ipoPlotOptions; }
+
+/**
+ * Verifies that the E Range is valid.
+ *
+ * @param prop QtProperty changed
+ * @param value Value it was changed to (unused)
+ */
+void IndirectSymmetriseView::verifyERange(QtProperty *prop, double value) {
+  UNUSED_ARG(value);
+
+  double eMin = m_dblManager->value(m_properties["EMin"]);
+  double eMax = m_dblManager->value(m_properties["EMax"]);
+
+  if (prop == m_properties["EMin"]) {
+    // If the value of EMin is negative try negating it to get a valid range
+    if (eMin < 0) {
+      eMin = -eMin;
+      m_dblManager->setValue(m_properties["EMin"], eMin);
+      return;
+    }
+
+    // If range is still invalid reset EMin to half EMax
+    if (eMin > eMax) {
```

Comment:
should be an `else if`, I believe there is a slight performance improvement compared to using just if's, so would still be worth doing

Reply:
For this we don;t want al else if since both should be checked.

Reply:
but if the first one is true, this bit of code will never run due to the return statement.

Reply:
True I didn't notice that. however in that case ther wouldn't be an impruvement since it will never reach the else. but I will change it for clarity.